aws/table_aws_wellarchitected_check_detail.go

```diff
@@ -0,0 +1,311 @@
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_wellarchitected_check_detail.listWellArchitectedCheckDetails", "error", err)
+		return nil, err
+	}
+
+	if len(answerList) == 0 {
+		return nil, nil
+	}
+
+	// Create session
+	svc, err := WellArchitectedClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_wellarchitected_check_detail.listWellArchitectedCheckDetails", "client_error", err)
+		return nil, err
+	}
+
+	// Unsupported region, return no data
+	if svc == nil {
+		return nil, nil
+	}
+
+	for _, answer := range answerList {
+		// List call
+		// Context can be cancelled due to manual cancellation or the limit has been hit
+		if d.EqualsQualString("pillar_id") != "" && d.EqualsQualString("pillar_id") != *answer.PillarId {
+			continue
+		}
+		if d.EqualsQualString("question_id") != "" && d.EqualsQualString("question_id") != *answer.QuestionId {
+			continue
+		}
+		_, err := fetchWellArchitectedCheckDetails(ctx, d, svc, answer)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_wellarchitected_check_detail.listWellArchitectedCheckDetails", "api_error", err)
+			return nil, err
+		}
+	}
+
+	return nil, nil
+}
+
+func fetchWellArchitectedCheckDetails(ctx context.Context, d *plugin.QueryData, svc *wellarchitected.Client, answer AnswerInfo) (interface{}, error) {
+
+	// Limiting the results
+	maxLimit := int32(50)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			maxLimit = limit
+		}
+	}
+
+	for _, choice := range answer.Choices {
+		if d.EqualsQualString("choice_id") != "" && d.EqualsQualString("choice_id") != *choice.ChoiceId {
+			continue
+		}
+
+		input := &wellarchitected.ListCheckDetailsInput{
+			MaxResults: maxLimit,
+			LensArn:    aws.String(*answer.LensArn),
+			PillarId:   aws.String(*answer.PillarId),
+			QuestionId: aws.String(*answer.QuestionId),
+			WorkloadId: aws.String(*answer.WorkloadId),
+			ChoiceId:   aws.String(*choice.ChoiceId),
+		}
+
+		paginator := wellarchitected.NewListCheckDetailsPaginator(svc, input, func(o *wellarchitected.ListCheckDetailsPaginatorOptions) {
+			o.Limit = maxLimit
+			o.StopOnDuplicateToken = true
+		})
+
+		for paginator.HasMorePages() {
+			output, err := paginator.NextPage(ctx)
+			if err != nil {
+				if strings.Contains(err.Error(), "ResourceNotFoundException") || strings.Contains(err.Error(), "ValidationException") {
+					return nil, nil
+				}
+				return nil, err
+			}
+
+			for _, item := range output.CheckDetails {
+				d.StreamListItem(ctx, CheckDetailInfo{item, answer.WorkloadId})
+
+				if d.RowsRemaining(ctx) == 0 {
+					return nil, nil
+				}
+			}
+		}
+	}
+	return nil, nil
+}
+
+func getAnswerDetailsForWorkload(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) ([]AnswerInfo, error) {
+	workload := h.Item.(types.WorkloadSummary)
+	workloadId := workload.WorkloadId
+
+	if d.EqualsQualString("workload_id") != "" && d.EqualsQualString("workload_id") != *workloadId {
```

Comment:
Can you please put a comment explaining the reason for returning nil in case of a mismatch in WorkloadID?